openapi/src/parts/workspace.yaml

```diff
@@ -100,6 +100,27 @@ paths:
         '500':
           $ref: '#/components/responses/ServerError'
 
+  /api/workspaces/v1/workspaceByUserFacingId/{workspaceUserFacingId}:
+    parameters:
+      - $ref: '#/components/parameters/WorkspaceUserFacingId'
+    get:
+      summary: Get an existing Workspace by userFacingId.
+      operationId: getWorkspaceByUserFacingId
+      tags: [Workspace]
+      responses:
+        '200':
```

Comment:
Aside, and maybe a question for @ddietterich: should we draw a distinction between listing workspaces and reading them, or is it our policy that you need read access to know it exists? ISTR it being the latter, which is consistent with my understanding of this new endpoint.

Reply:
Regardless of this API, you will be able to determine this by trying to create a workspace with a given UFID to see if it is rejected.

Reply:
In fact, I was planning to use this API in the "Create Workspace" flow to determine if a provided UFID is taken.

So I was expecting this API to return the UUID, not the full object.  I think before it returns the full object, you would have to have access to use it?

Reply:
`getWorkspaceByUserFacingId` will be used for:

- UI: User loads workspace page. Url has userFacingId
- CLI ` terra workspace set --id=userFacingId`

`getWorkspaceByUserFacingId` will only return the workspace if Sam says you have READ access.

For UI determining "is UFID taken", I was thinking that would be a separate endpoint: PF-1605. Something like `/api/workspaces/v1/isUserFacingIdTaken/{workspaceUserFacingId}`. @cbookg do you want separate endpoint, or do you want to call `getWorkspaceByUserFacingId` and 403/404 means ufId is taken?

Reply:
For Jay's question: we only list workspaces that the caller has some access to. We do this by first doing a Sam call to get the list of WORKSPACE objects the caller has access to and then filtering our DAO query with that list.

Reply:
> For UI determining "is UFID taken", I was thinking that would be a separate endpoint: PF-1605. Something like /api/workspaces/v1/isUserFacingIdTaken/{workspaceUserFacingId}. @cbookg do you want separate endpoint, or do you want to call getWorkspaceByUserFacingId and 403/404 means ufId is taken?

Either option is fine with me.  If it is more efficient for the backend to check existing of an ID than to populate the full get result, a distinct ID might be better.

Reply:
I vote for fewer endpoints. The lookup is a single row in the workspace table and the Sam access check, so returning the full result is not a big cost.

Reply:
SG. Closed PF-1605 and updated PF-1606.